sockeye/layers.py

```diff
@@ -732,3 +796,126 @@ def hybrid_forward(self, F, data, steps, weight):  # pylint: disable=arguments-d
             data = data * (self.num_embed ** 0.5)
 
         return F.broadcast_add(data, pos_embedding)
+
+
+class SSRU(AutoregressiveLayer):
+    """
+    Simpler Simple Recurrent Unit (Kim et al, 2019)
+
+    Variant of an LSTM cell aimed at reducing computational dependency across time steps.
+    Formally described as:
+
+    (1) f[t] = sigmoid(W1[t] * x[t] + b[t])
+    (2) c[t] = f[t] . c[t-1] + (1 - f[t]) . W2[t] * x[t]
+    (3) h = ReLU(c[t])
+
+    where:
+        . represents elementwise multiplication;
+        x[t] is the input at time step t;
+        f[t] is the output of the forget gate at time step t;
+        c[t] is the cell state at time step t;
+        h is the output of the unit.
+
+    :param model_size: number of hidden units
+    :param inference_only: flag used to indicate execution at inference time
+    :param prefix: prefix prepended to the names of internal Symbol instances
+    :param dtype: data type
+    """
+    def __init__(self,
+                 model_size: int,
+                 inference_only: bool,
+                 prefix: str = SSRU_PREFIX,
+                 dtype: str = C.DTYPE_FP32) -> None:
+        super(SSRU, self).__init__(prefix=prefix)
+
+        self.model_size = model_size
+        self.inference_only = inference_only
+
+        self.cell_state_transform = self._inference_cell_state_transform \
+                                    if inference_only else self._training_cell_state_transform
+
+        with self.name_scope():
+            self.forget_gate = quantization.QuantizableDense(in_units=model_size,
+                                                             units=model_size,
+                                                             activation="sigmoid",
+                                                             flatten=False,
+                                                             prefix="forget_gate_",
+                                                             dtype=dtype)
+
+            self.linear = quantization.QuantizableDense(in_units=model_size,
+                                                        units=model_size,
+                                                        use_bias=False,
+                                                        flatten=False,
+                                                        prefix="linear_",
+                                                        dtype=dtype)
+
+    @property
+    def prefix(self) -> str:
```

Comment:
this does overwrite the `HybridBlock.prefix` attribute, does it? Fine with me, especially because MXNet 2.0 won't have Block prefixes anymore.

Reply:
Good point, it does overwrite that.